conductor-client/src/main/java/com/openlattice/edm/properties/PostgresTypeManager.java

```diff
@@ -125,13 +125,13 @@ public PostgresTypeManager( HikariDataSource hds ) {
         try ( Connection connection = hds.getConnection();
                 PreparedStatement ps = connection.prepareStatement( getPropertyTypes ) ) {
             List<PropertyType> result = Lists.newArrayList();
-            ResultSet rs = ps.executeQuery();
-            while ( rs.next() ) {
-                result.add( ResultSetAdapters.propertyType( rs ) );
+            try ( ResultSet rs = ps.executeQuery(); ) {
+                while( rs.next() ) {
+                    result.add( ResultSetAdapters.propertyType( rs ) );
+                }
+                connection.close();
+                return result;
```

Comment:
All of these all look nearly identical. Is it possible to extract a common function to handle the try + while block of logic? In JS, we can do something like this (pseudocode):

```
function close(ps, callback) {
  try {
    ResultSet rs = ps.executeQuery();
    while( rs.next() ) {
      callback(rs);
    }
    connection.close();
  }
  catch () { ... }
  finally { ... } 
}

function getPropertyTypes() {
  ...
  PreparedStatement ps = ...
  List<PropertyType> result = ...
  close(ps, (rs) => {
    result.add( ResultSetAdapters.propertyType( rs ) );
  });
  return result;
}

```

Reply:
Yeah, can definitely do that. I was just trying to spend as little time on this code as possible right now because its quite old and should likely be replaced with Kotlin updated code

Reply:
( + should someday be replaced by hazelcast-only calls instead of talking to postgres directly)